Documentation/CraftingNuGetPackages.md

```diff
@@ -0,0 +1,69 @@
+# Crafting a NuGet Package for MSB4U
+
+> MSBuildForUnity consumes .NET based packages without any modifications to the package. This documentation describes how C++, WinRT or packages with Unity-specific contents can easily be adapted to work with MSBuildForUnity.
+
+## Quickstart Summary
+
+Through the use of a [Build Targets or Props](https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#include-msbuild-props-and-targets-in-a-package) file inside a NuGet package, we can leverage properties provided by MSBuildForUnity to appropriately select the contents to bring into the Unity project. The following snippet is taken from [Microsoft.Windows.MixedReality.DotNetWinRT](https://www.nuget.org/packages/Microsoft.Windows.MixedReality.DotNetWinRT/) NuGet package. It brings in the `Unity` folder packaged in the root of the NuGet package to the output folder inside Unity.
+
+```xml
+<Project>
+  <PropertyGroup>
+    <!--All extra resources for this package will be placed under this folder in the output directory.-->
+    <PackageDestinationFolder>$(MSBuildThisFileName)</PackageDestinationFolder>
+  </PropertyGroup>
+  
+  <ItemGroup Condition="'$(MSBuildForUnityVersion)' != ''">
```

Comment:
Do you know if there's any way to highlight/comment the MsBuildForUnity specific things within here, vs the things that are just general MSBuild things? As a person who also isn't doing msbuild things on a day to day basis, it would be helpful to have some indicator about what is unique to MsBuildForUnity vs. simple msbuild stuff (i.e. the structure of <Link>$(PackageDestinationFolder)\%(RecursiveDir)%(Filename)%(Extension)</Link> - seems like it's general msbuild stuff, but MSBuildThisFileDirectory..\Unity\** sounds interesting - is there something magical about the .. that's interesting to talk about, etc)